### DISCUSSION TITLE
calc.derivs = FALSE and REML = FALSE

### DISCUSSION THREAD

**Discussion by tatiana-pashkova** [RePsychLing/SMLP2023#24] [Q&A] [answered]
Yesterday, Reinhold mentioned that we should always use control = lmerControl(calc.derivs = FALSE)) and REML = FALSE. Could you explain what these things mean and if there is a reference for them, for example, if a reviewer asks me why I used control = lmerControl(calc.derivs = FALSE)?

**Comment by dmbates** [comment, marked as answer]
The REML criterion is a hold-over from earlier days when people felt that estimating "variance components" as variances would naturally imply that these estimators were normally, or at least symmetrically, distributed.  If that were the case then the mean of the distribution would be a good location measure and we would want to mean of the estimator to be close to the parameter being estimated.  This is what an "unbiased estimator" means.

They played with the definition of the likelihood to define "residual" or "restricted" likelihood and these REML estimates, which are not exactly unbiased for these models, but are closer to being unbiased.

However, the whole question is moot because the estimators are quite skewed and the mean is not a good measure of location.  The likelihood and likelihood-ratio tests are more cleanly defined for these models.  We characterize variability according to multiple evaluations of the model at different parameter values through, e.g. profiling, instead of trying to do only one fit and a bunch of mathematical approximations.

As for `calc.derivs = FALSE`, the evaluation is using numerical (finite-difference) approximations to both the first and second-order partial derivatives to evaluate whether convergence has been achieved.  First, these approximations are notoriously inaccurate and second the convergence tests don't apply for convergence to a singular covariance, which is when they would be most useful.  In the end the information is inaccurate and of little use but expensive to produce.

**Comment by palday** [comment]
I have a few slides on this from several years ago here: https://rpubs.com/palday/lme4-singular-convergence

I hope that they still make a little sense without me talking in the background.
